### PR TITLE
Vision Transformer for ADNI Dataset

### PR DESCRIPTION
**Pull request by kunsi81**
This pull request is an implementation of a Vision Transformer trained on the ADNI dataset.

The trained model acheived a test accuracy of 66.9%, which unfortunately does not meet the requirement of 80%. 

The following files are included in this pull request under `recognition/46975115_ViT`:
- `figures&images` file: contains figures and images for the `README.md` file
- `README.md`: Description of algorithm and the problem that it solves, how it works in a paragraph and a evidence of training and results.
- `dataset.py`: Preprocessing and loading of the dataset.
- `modules.py`: Implementation of the vision transformer model.
- `predict.py`: Generate results for the trained model.
- `train.py`: Training of the model, provides plots of accuracy/loss and test accuracy.
- `utils.py`: Stores hyperparameters and optimizer function.

### CODE REVIEW

**Comment by prprakash02** [comment]
**This is an initial inspection**

**1) Recognition Problem:**

a) Algorithm solves the respective pattern recognition problem appropriately: **(+3)**
b) Algorithm implemented in PyTorch demonstrated and functions as intended to solve the respective problem: **(+3)**
c) Good design implemented in PyTorch that solves the problem: **(+1)**
d) Commenting: **(+1)**
e) Algorithm is normal or above difficulty while implemented in TF/PyTorch while solving the respective problem: Hard level **(+5)**
f) Algorithm is hard difficulty while implemented in TF/PyTorch while solving the respective problem: Hard level **(+5)**

**2) Commit Log:**

a) Meaningful commit messages with evidence of individual work and originality: **(+2)**
b) Progressive commits and logical log structure providing evidence of individual work and originality: **(+3)**

**3) Documentation:**

Requirement:

Tittle: Done
Description of the algorithm: Done
Problem that it solves: Done
How it works in a paragraph (example any pre-processing): Done
Figure/visualisation (any input, output & results): Done
Dependencies: Done

Marking:

a) Description and explanation of the working principles of the algorithm implemented and the problem it solves: **(+5)**
b) Description of usage and comments throughout scripts: **(+3)**
c) Proper formatting using GitHub markdown: **(+2)**


**4) Pull Request**

a) Creating a pull-request into correct branch with a working and demonstrable version of the algorithm: **(+2)**
b) Incorporating feedback into the pull request: **(+2)**
c) Description of and comments within the pull request: **(+1)**

**Feedback:**
Good attempt, could have tried using better data augmentations and a variational model for better generalization

**Any Changes Needed: No**

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request includes a clear description about the file structure .


File Organizing: Well-organized files.

Commit Log:
Commit messages are  progressive for the Recognition Problem solved using 4 files .
Commits  demonstrate  logical development flow.
Commits are regularly made, showing incremental development rather than bulk updates.

Documentation:
The README file should contain **Conclusion and Future Improvements** .
Code comments and docstrings are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. </td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard. ViT (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time submission.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-3</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_
